### PR TITLE
add result to compare navigation, add exact search

### LINKED ISSUE #26: make search results "selectable" and "comparable"
Search results should be "selectable" via checkboxes or some selection mechanism, after which the selected results should be forwarded to the drug label diff view made by Sam.

### PR DESCRIPTION
**Pull request by ky-soren**
This PR implements "Search Result -> Compare" navigation. PR also adds exact search as well as some search speed improvements. NOTE: Search highlighting does not work until I figure out how to return the correct section

### CODE REVIEW

File: dle/search/services.py [hunk 1]
```diff
@@ -27,40 +27,58 @@ def validate_search(request_query_params_dict: QueryDict) -> SearchRequest:
...
     for k, v in search_request_dict.items():
         if v and k in search_filter_mapping:
             param_key = search_filter_mapping[k]
             if k == "select_section" and not v:
                 continue
             sql_params[param_key] = v
-            additional_filter = f"AND LOWER({param_key}) = LOWER(%({param_key})s) "
+            additional_filter = f"AND LOWER({param_key}) = %({param_key})s "
             raw_sql += additional_filter
     raw_sql += "LIMIT 30" #can remove this once we're done testing
```
Comment (kenbr7613) [inline dle/search/services.py:81]: Is this the section limit you were worried about? Can you change it to limit 125?

Comment (ky-soren) [reply dle/search/services.py:81]: no that's just to limit the search results to the top 30. The section limit is just hardcoded at the moment in the `type_ahead_mapping`

Comment (kenbr7613) [reply dle/search/services.py:81]: Ok. You can probably remove it now
```
mysql> SELECT count(DISTINCT section_name) FROM data_productsection;
+------------------------------+
| count(DISTINCT section_name) |
+------------------------------+
|                           97 |
+------------------------------+
```

**Review by samNson — APPROVED** [review]
Changes in this PR look good to me. I'll merge this PR per the discussion this morning.

**Pull request merged**